tests/online_ddl/run.sh

```diff
@@ -49,12 +48,11 @@ function real_run() {
 	# manually create index to pass check_sync_diff
 	run_sql_tidb "show create table online_ddl.t_target"
 	check_not_contains "KEY \`name\`"
-	run_sql_tidb "alter table online_ddl.t_target add key name (name)"
+	# TODO why need this?
```

Comment:
sync-diff will report "different table structure"

Reply:
root cause located: binlog filter don't filter sql

input sql is:
```
ALTER TABLE `online_ddl`.`t1` ADD INDEX `name`(`name`)`
```
* paser return this sql type is `AlterTableStmt`, but we set filter event type is `create index`
* filter regular expressions:`"ALTER\\s.*TABLE[\\s\\S]*ADD\\s+KEY\\s+NAME"` also does not match this sql.

Reply:
I remember in query events the sql is written as-is, so it should be `alter table t2 add key name (name);` rather than your sql.

Reply:
maybe caused by you delete this filter logic in https://github.com/pingcap/dm/pull/1824/files#diff-d3b44c58774ef21a03b59014f103c9ff35671c9d131ce9dc3d6b480e9ce0004dL40

don't change the logic if you don't totally understand them.

Reply:
* the input raw is `ALTER TABLE `online_ddl`.`_t2_new` add key name (name)`, I just found it by print sql  after`case *replication.QueryEvent:`

* I deliberately deleted the repetitive `skipQuery` because i am refarctoring hande ddl logic and i don't want to call `skipQuery` twice

* The main refactoring plan is [here](https://github.com/Ehco1996/dm/blob/0d1888456680c44b6aaa7bc908597fd80d307445/syncer/syncer.go#L1919) I deliberately deleted the repetitive logic


thie PR is still **WIP**,  I will ask for your help to review it when it is finished

Reply:
the reason why we filter SQL in two place is: if the SQL can't be parsed, it should support filtering by regular expression in a very early stage. But if it can be parsed, some transformation is needed to decide if it's should be filtered.

Also it's better to find a way to unify them.

You could verify the reasons of some code with us before made changes, to reduce unnecesary work.

Reply:
DON'T mark my comments as resolved, I don't think it's resolved